src/main/java/org/opentripplanner/middleware/controllers/api/ApiController.java

```diff
@@ -197,29 +198,26 @@ private String deleteOne(Request req, Response res) {
      * Convenience method for extracting the ID param from the HTTP request.
      */
     private T getObjectForId(Request req, String id) {
-        T object = persistence.getById(id);
-        if (object == null) {
-            logMessageAndHalt(
-                req,
-                HttpStatus.NOT_FOUND_404,
-                String.format("No %s with id=%s found.", classToLowercase, id),
-                null
-            );
-        }
-        return object;
+        return checkAndLogIfNull(req, persistence.getById(id), "id", id);
     }
 
     /**
      * Convenience method for extracting the attribute/field param from the HTTP request.
      */
-    private T getObjectWithField(Request req, String field, String value) {
-        T object = persistence.getByField(field, value);
+    private T getFirstObjectByFieldValue(Request req, String field, String value) {
+        return checkAndLogIfNull(req, persistence.getOneFiltered(eq(field, value)), field, value);
+    }
+
+    /**
+     * Log if an object was not found for a request.
+     */
+    private T checkAndLogIfNull(Request req, T object, String field, String value) {
```

Comment:
This is too much abstraction to save just a little bit of repeated code. It makes things less readable, so I think the original way of doing this in `getObjectForId` is preferred.

Reply:
Or maybe a hybrid approach would be better where you can change this method to be something like:

```java
private void haltIfNull(T object, Request req, String message) {
    if (object == null) {
        logMessageAndHalt(req, HttpStatus.NOT_FOUND_404, message);
    }
}
```